compose/production/traefik/traefik.yml

```diff
@@ -0,0 +1,85 @@
+log:
+  level: INFO
+
+entryPoints:
+  web:
+    # http
+    address: ":80"
+
+  web-secure:
+    # https
+    address: ":443"
+
+  elasticsearch:
+    address: ":9200"
+
+certificatesResolvers:
+  letsencrypt:
+    # https://docs.traefik.io/master/https/acme/#lets-encrypt
+    acme:
+      email: "flavio.f.martins@tecnico.ulisboa.pt"
+      storage: /etc/traefik/acme/acme.json
+      # https://docs.traefik.io/master/https/acme/#httpchallenge
+      httpChallenge:
+        entryPoint: web
+
+http:
+  routers:
+    web-router:
+      rule: "Host(`pe.inesc-id.pt`) || Host(`www.pe.inesc-id.pt`)"
```

Comment:
O computador não tem autoridade sobre os seus subdominios.
Type | Domain Name | IP Address | TTL
-- | -- | -- | --
A | pe.inesc-id.pt | 146.193.39.117

Reported by inesc-id.inesc-id.pt

  | DNS Record Published | DNS Record not found | More Info
-- | -- | -- | --

Reported by inesc-id.inesc-id.pt

Levantando um erro de certificados:
```
Unable to obtain ACME certificate for domains \"pe.inesc-id.pt,elasticsearch.pe.inesc-id.pt\": unable to generate a certificate for the domains [pe.inesc-id.pt elasticsearch.pe.inesc-id.pt]: error: one or more domains had a problem:\n[elasticsearch.pe.inesc-id.pt] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up A for elasticsearch.pe.inesc-id.pt - check that a DNS record exists for this domain; DNS problem: NXDOMAIN looking up AAAA for elasticsearch.pe.inesc-id.pt - check that a DNS record exists for this domain\n" ACME CA="https://acme-v02.api.letsencrypt.org/directory" providerName=letsencrypt.acme routerName=elasticsearch-secure-router@file rule="Host(`pe.inesc-id.pt`) || Host(`elasticsearch.pe.inesc-id.pt`)
```

De qualquer modo diria que os certificados deviam ser controlados fora do controlo deste projeto. Dado que nem sequer sabemos qual vai ser o endereço final da app.

Reply:
Talvez fosse bom ter uma pasta etc/httpd/conf.d/ para incluir neste repo o config file do Apache com os proxypass, etc.